custom_components/tahoma/cover.py

```diff
@@ -269,13 +228,12 @@ def device_state_attributes(self):
     @property
     def icon(self):
         """Return the icon to use in the frontend, if any."""
+        icon = None
         if self._lock_timer > 0:
+            icon = "mdi:lock-alert"
             if self._lock_originator == "wind":
-                return "mdi:weather-windy"
-            else:
-                return "mdi:lock-alert"
-
-        return self._icon
+                icon = "mdi:weather-windy"
+        return icon
```

Comment:
I would prefer to have this without the icon variable, like we do in `binary_sensor` and `sensor`. Where we directly return the value that is appropriate. 

```
if self._lock_timer > 0:
   if self._lock_originator == "wind":
       return "mdi:weather-windy"

   return "mdi:lock-alert"

return None
```

Reply:
OK, I will keep in mind this rule. Fixed.